File: python/matrix/matrix.py
class Matrix(object):
    def __init__(self, matrix_string):
        self.data = list(map(__row_to_int__, matrix_string.split("\n")))
 
    def row(self, index):
        return self.data[index - 1]

    def column(self, index):
        column = []
        for row in self.data:
            column.append(row[index - 1])

        return column

def __row_to_int__(row):
    return list(map(int, row.split()))

File: javascript/matrix/matrix.js
const fromRowToNumbers = row => row.split(' ').map(elem => Number(elem));

const transposeMatrix = matrix => matrix[0].map((_, i) => {
  return matrix.map(row => row[i]);
});


export class Matrix {
  constructor(matrix) {
    this.data = matrix.split('\n').map(fromRowToNumbers);
    this.transposed = transposeMatrix(this.data);
  }

  get rows() {
    return this.data;
  }

  get columns() {
    return this.transposed;
  }
}


